app/jenkins-for-jira-ui/src/components/SharePage/SharePage.tsx

```diff
@@ -0,0 +1,154 @@
+import { CONFIG_PAGE } from '../../common/constants';
+
+const analyticsClient = new AnalyticsClient();
+
+export const getSiteNameFromUrl = (url: string): string => {
+	try {
+		const urlObject = new URL(url);
+		return urlObject.hostname;
+	} catch (error) {
+		console.error('Error extracting site name:', error);
+		return '';
+	}
+};
+
+export const getSharePageMessage = (globalPageUrl: string, moduleKey?: string): string => {
+	const versionRequirementMessage = moduleKey === CONFIG_PAGE
+		? `
+Can’t see a page when you follow this link? Ask your Jira admin to update Jenkins for Jira at:
+
+Apps > Manage your apps > Jenkins for Jira (Official)` : '';
+
+	return `Hi there,
+Jenkins for Jira is now installed and connected on
+
+${getSiteNameFromUrl(globalPageUrl)}.
+
+To get data flowing from Jenkins to Jira, follow the set up guide(s) on this page:
+
+${globalPageUrl}
+
+You'll need to follow the guide for each server connected.
+${versionRequirementMessage}`;
+};
+
+type SharePageProps = {
+	analyticsUIScreenNameEnum: AnalyticsUiEventsEnum;
+	buttonAppearance?: string;
+};
+
+export const SharePage = ({
+	analyticsUIScreenNameEnum,
+	buttonAppearance
+}:SharePageProps): JSX.Element => {
+	const textAreaRef = useRef<HTMLTextAreaElement>(null);
+	const [isCopiedToClipboard, setIsCopiedToClipboard] = useState(false);
+	const isMountedRef = React.useRef<boolean>(true);
+	const [globalPageUrl, setGlobalPageUrl] = useState<string>('');
+	const [showSharePage, setShowSharePage] = useState<boolean>(false);
+	const buttonStyleAppearance = buttonAppearance === 'primary' ? 'primary' : 'default';
+
+	useEffect(() => {
+		const fetchData = async () => {
+			try {
+				const url = await fetchGlobalPageUrl();
+				setGlobalPageUrl(url);
+			} catch (error) {
+				console.error('Error fetching data:', error);
+			}
+		};
+
+		fetchData();
+
+		return () => {
+			// Cleanup function to set isMountedRef to false when the component is unmounted
+			isMountedRef.current = false;
+		};
+	}, []);
+
+	const handleShowSharePageModal = async () => {
+		await analyticsClient.sendAnalytics(
+			AnalyticsEventTypes.UiEvent,
+			AnalyticsUiEventsEnum.SharePageName,
+			{
+				source: analyticsUIScreenNameEnum,
+				action: `clicked - ${AnalyticsUiEventsEnum.SharePageName}`,
+				actionSubject: 'button'
+			}
+		);
+		setShowSharePage(true);
+	};
+
+	const handleCloseShowSharePageModal = async () => {
+		setShowSharePage(false);
+	};
+
+	const sharePageMessage = getSharePageMessage(globalPageUrl);
+
+	const handleCopyToClipboard = async () => {
+		await analyticsClient.sendAnalytics(
+			AnalyticsEventTypes.UiEvent,
+			AnalyticsUiEventsEnum.CopiedToClipboardName,
+			{
+				source: analyticsUIScreenNameEnum,
+				action: `clicked - ${AnalyticsUiEventsEnum.CopiedToClipboardName}`,
+				actionSubject: 'button'
+			}
+		);
+
+		if (textAreaRef.current) {
+			textAreaRef.current.select();
+			document.execCommand('copy');
```

Comment:
Forge apps do support copy to clipboard, so maybe use that instead of using JS here. ([source](https://community.developer.atlassian.com/t/using-navigator-object-in-forge-app/60755/3))